odxtools/nameditemlist.py

```diff
@@ -115,13 +149,13 @@ def __eq__(self, other: object) -> bool:
         if not isinstance(other, NamedItemList):
             return False
         else:
-            return self._list == other._list
+            return self._names == other._names and self._values == other._values and self._item_to_name_fn == other._item_to_name_fn
 
-    def __iter__(self):
-        return iter(self._list)
+    def __iter__(self):  # -> Iterator[T]: # <- this leads to *many* type checking errors
+        return iter(self._values)
```

Comment:
this will be confusion to some users, it have a dict interface, but iterate like a list

Reply:
the class is primarily intended to be usable as a normal list with some extra features. (i.e., direct item access via attributes. That's why it's called `NamedItemList`, not `NamedItemDict`. ;) )